dev/refinement-bot/src/getRandomIssue.ts

```diff
@@ -0,0 +1,62 @@
+import { setOutput, setFailed } from '@actions/core'
+import { getOctokit } from '@actions/github'
+import { Repository } from '@octokit/graphql-schema'
+import type { Endpoints } from '@octokit/types'
+import dotenv from 'dotenv'
+dotenv.config()
+
+export type Events = Endpoints['GET /repos/{owner}/{repo}/labels']['response']['data']
+
+console.assert(process.env.GITHUB_TOKEN, 'GITHUB_TOKEN not present')
+console.assert(process.env.TEAM, 'TEAM not present')
+
+// get valus
+const octokit = getOctokit(process.env.GITHUB_TOKEN as string)
+const team = process.env.TEAM as string
+
+const getIssues = async (): Promise<void> => {
+    // step 1 get estimates;
+    const result = await octokit.graphql<{ repository: Repository }>(`{
+      repository(owner:"sourcegraph", name:"sourcegraph") {
+        labels(first: 100, query: "estimate/") {
+          nodes {
+            name
+          }
+        }
+      }
+    }`)
+    // generates a comma seperates list of every estiamte label
+    let estimates: string | undefined
+    try {
+        estimates = result.repository.labels?.nodes?.map(item => item?.name).join(',')
+        if (!estimates) {
+            throw new Error('missing estimates')
+        }
+    } catch {
+        setFailed('unable to get estimate labels')
+        return
+    }
+
+    // step 2 get issues without estimates that are on ${team}
+    const query = `repo:"sourcegraph/sourcegraph" is:issue is:open label:${team} -label:${estimates} sort:updated-desc no:assignee -label:tracking`
```

Comment:
There's a possible bug where `estimates` could contain undefined which would result in stuff like: ",test,,", I think?

I like this version below. The `isDefined` helper is just something which tells `.map` that `item.name` definitely exists now. We use it in a few places

```suggestion
    const isDefined = <T>(value: T): value is NonNullable<T> => value !== undefined && value !== null
    const estimates = result.repository.labels?.nodes?.filter(isDefined).map(item => item.name) || []

    if (estimates.length === 0) {
        setFailed('unable to get estimate labels')
        return
    }
    // step 2 get issues without estimates that are on ${team}
    const query = `repo:"sourcegraph/sourcegraph" is:issue is:open label:${team} -label:${estimates.join(
        ','
    )} sort:updated-desc no:assignee -label:tracking`
```